private/bufpkg/bufplugin/bufplugindocker/auth.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufplugindocker
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"strings"
+)
+
+type RegistryAuth struct {
```

Comment:
Got it, if we end up keeping this concrete struct because it matches the Docker API, let's add a reference and annotate this as such.

```suggestion
// RegistryAuth represents the fields required to authenticate with the Docker Engine API.
// Ref: https://docs.docker.com/engine/api/v1.41/#section/Authentication
type RegistryAuth struct {
```

Also, if this isn't moved into the `bufplugindocker.go` file, let's rename the file to match the `type` name (i.e. `registry_auth.go`).

Reply:
We might also want to rename this to `RegsitryAuthConfig` since it really is just a configuration that can map to a different representation. More details in a comment on the call-site below.